crates/flox/src/commands/channel.rs

```diff
@@ -132,62 +156,229 @@ fn construct_search_params(search_term: &str, flox: &Flox) -> Result<SearchParam
+        // Build a collection of packages and which inputs they are seen in
+        package_to_inputs
+            .entry(d.package.clone())
+            .and_modify(|inputs| {
+                inputs.insert(d.input.clone());
+            })
+            .or_insert_with(|| {
+                let mut inputs = HashSet::new();
+                inputs.insert(d.input.clone());
+                inputs
+            });
+    }
+
+    // For any package that comes from more than one input, mark it as needing to be joined
+    for d in display_items.iter_mut() {
+        if let Some(inputs) = package_to_inputs.get(&d.package) {
+            d.join = inputs.len() > 1 || d.join;
+        }
+    }
+
+    let mut deduped_display_items = Vec::new();
+    for d in display_items.into_iter() {
+        if let Some(inputs) = package_to_inputs.get_mut(d.package.as_str()) {
+            // Remove this input so this (input, package) pair is never seen again
+            if inputs.remove(&d.input) {
+                deduped_display_items.push(d.clone());
+            }
+            if inputs.is_empty() {
+                package_to_inputs.remove(&d.package);
+            }
+        }
+    }
+
+    deduped_display_items
+}
+
+/// Show detailed package information
+#[derive(Bpaf, Clone)]
+pub struct Show {
+    /// The package to show detailed information about. Must be an exact match
+    /// for a package name e.g. something copy-pasted from the output of `flox search`.
+    #[bpaf(positional("search-term"))]
+    pub search_term: String,
+}
+
+impl Show {
+    pub async fn handle(self, flox: Flox) -> Result<()> {
+        let search_params = construct_show_params(&self.search_term, &flox)?;
+
+        let (search_results, exit_status) = do_search(&search_params)?;
+
+        if search_results.results.is_empty() {
+            bail!("no packages matched this search term: {}", self.search_term);
+        }
+        // Render what we have no matter what, then indicate whether we encountered an error.
+        // FIXME: We may have warnings on `stderr` even with a successful call to `pkgdb`.
+        //        We aren't checking that at all at the moment because better overall error handling
+        //        is coming in a later PR.
+        render_show(search_results.results.as_slice())?;
+        if exit_status.success() {
+            Ok(())
+        } else {
+            bail!(
+                "pkgdb exited with status code: {}",
+                exit_status.code().unwrap_or(-1),
+            );
+        }
+    }
+}
+
+fn construct_show_params(search_term: &str, flox: &Flox) -> Result<SearchParams> {
+    let parts = search_term
+        .split(SEPARATOR)
+        .map(String::from)
+        .collect::<Vec<_>>();
+    let (input_name, package_name) = match parts.len() {
+        1 => Ok((None, Some(parts[0].clone()))),
+        2 => Ok((Some(parts[0].clone()), Some(parts[1].clone()))),
+        _ => Err(ShowError::InvalidSearchTerm(search_term.into())),
+    }?;
```

Comment:
```suggestion
    let parts = search_term.split(SEPARATOR).collect::<Vec<_>>();

    let (input_name, package_name) = match parts[..] {
        [package_name] => (None, Some(package_name.to_owned())),
        [input_name, package_name] => (Some(input_name.to_owned()), Some(package_name.to_owned())),
        _ => Err(ShowError::InvalidSearchTerm(search_term.into()))?,
    };
```

somewhat more rusty?

Reply:
Fixed